truckersmp-cli

```diff
@@ -28,6 +28,8 @@ import urllib.request
 from getpass import getuser
 from gettext import ngettext
 
+import vdf
```

Comment:
Why do you think it's best to use it as submodule? Is that a common thing with python?
"[…] because it doesn't depend on other packages" isn't a good explanation to me.

I thought the correct way for dependencies is to install them alongside (depend) our script either from a [repository](https://repology.org/project/python:vdf/versions) or [pip](https://pypi.org/project/vdf/) and expect it to be in pythons search path.
The advantage over a submodule is we don't have to keep track of updates until something breaks and we don't include things which might already are installed on the system or could potentially clash with other packages which provide python-vdf.
Here are a few example python projects which depend on python-vdf and don't provide it in a submodule: [[1][1],[2][2],[3][3],[4][4]]

[1]: https://github.com/Matoking/protontricks/blob/351474c5b2d4bf1d60bf201a2128848d0f5c38df/src/protontricks/steam.py#L11
[2]: https://github.com/gamer-os/steam-tweaks/blob/master/steam-config#L4
[3]: https://github.com/coldfix/steam-acolyte/blob/11b1b2df44a322879976ea062dadfd427a3410b6/steam_acolyte/steam.py#L4
[4]: https://github.com/ValvePython/steam/blob/2267f9cc2a5e9a1ca0ecb1401efaa98469437c1a/steam/client/builtins/apps.py#L1

Reply:
Example 1, 3, and 4 are supposed to be installed via `pip`...

Sorry, I didn't know that you're considering packaging/uploading our software *for PyPI*.

Then, we only need to write `vdf` to `setup.py` (or `requirements.txt`) for dependencies:

* https://github.com/ValvePython/steam/blob/master/setup.py
* https://github.com/Matoking/protontricks/blob/master/requirements.txt
* https://github.com/ValvePython/steam/blob/master/requirements.txt

> I thought the correct way for dependencies is to install them alongside (depend) our script either from a [repository](https://repology.org/project/python:vdf/versions) or [pip](https://pypi.org/project/vdf/) and expect it to be in pythons search path.

That's right. If we're going to publish our software to PyPI, there is no reason to use submodule here, but we need to change the following sections in `README.md`:

* [Dependencies > Required](https://github.com/lhark/truckersmp-cli/#required) (Add `(python3-)pip` and `vdf`)
* [Install](https://github.com/lhark/truckersmp-cli/#install) (Replace current content with new installation instruction after publishing)

---

A problem for me is that I need to install `vdf` via `pip` beforehand manually (Debian-based systems don't provide this). Otherwise the script won't work. Some users might even dislike to install packages via `pip` or install `pip` itself.

One solution is to handle `ImportError`, the autodetection feature would be used only when `vdf` is available. In this case, `vdf` would be [optional dependency](https://github.com/lhark/truckersmp-cli/#optional).

```python
vdf_is_available = False
try:
    import vdf
    vdf_is_available = True
except ImportError:
    pass
	
# ...

    if not vdf_is_available:
        return None
```

Reply:
> One solution is to handle `ImportError`, the autodetection feature would be used only when `vdf` is available. In this case, `vdf` would be [optional dependency](https://github.com/lhark/truckersmp-cli/#optional).

Guess that's the best thing to do here. It's an optional convenience function anyway.

> we need to change the following sections in `README.md`

Yes, let's revisit that combined with all the other things before the first release, I'm adding this to the todo-list.